### PR TITLE
Publish liminal packages and release notes

### PR DESCRIPTION
**Pull request by naturalett**
@assapin I started working on [CI for github](https://issues.apache.org/jira/browse/LIMINAL-16)

### CODE REVIEW

**Comment by aviemzur** [comment]
R: @assapin

File: .github/workflows/publish-package.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  publish-package:
+    name: Publish package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set env
+        run: echo "LIMINAL_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Set up Python
```
Comment (assapin) [inline .github/workflows/publish-package.yml:16]: add git tag

File: .github/workflows/publish-package.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+          TWINE_REPOSITORY_URL: https://upload.pypi.org/legacy/
+        run: |
+          python setup.py sdist bdist_wheel
```
Comment (assapin) [inline .github/workflows/publish-package.yml:30]: @naturalett should we add a flavor which pushes to test pypi?

**Review by assapin — CHANGES_REQUESTED** [review]

**Comment by assapin** [comment]
@naturalett I think we need these steps for pypi:

1.have the code to contain a version.py file with the version id we are building
3. Build
4. push to the right pypi repo - either test or prod based on version name or manually choose

File: .github/workflows/publish-package.yml (hunk 1 above)
Comment (naturalett) [reply .github/workflows/publish-package.yml:16]: The trigger starts by git tag.
This is my condition:
```
on:
  push:
    tags:
      - '**'
```

File: .github/workflows/publish-package.yml (hunk 2 above)
Comment (naturalett) [reply .github/workflows/publish-package.yml:30]: 1. Would it be better to push to test pypi for every push to master?
2. To push to pypi will be by a user demand.

WDYT?

@assapin

Comment (assapin) [reply .github/workflows/publish-package.yml]: test pypi on merge to master seems logical
"real" pypi is only for RC and official versions - triggered manually makes sense too for now

**Comment by assapin** [comment]
I think it would benefit if we created first an understanding of the logical flow before implementing.
lets set time this week to properly decide on the minimal flows we want to implement

File: tests/runners/airflow/build/http/python/test_python_server_image_builder.py [hunk 3]
```diff
@@ -50,9 +50,9 @@ def test_build_python_server(self):
     def test_build_python_server_with_pip_conf(self):
         build_out = self.__test_build_python_server(use_pip_conf=True)
 
-        self.assertTrue(
```
Comment (assapin) [inline tests/runners/airflow/build/http/python/test_python_server_image_builder.py:53]: did you mean to comment this out? why?

File: tests/runners/airflow/tasks/test_python.py [hunk 4]
```diff
@@ -59,14 +76,14 @@ def test_apply_task_to_dag(self):
 
         for task in dag.tasks:
             print(f'Executing task {task.task_id}')
-            task.execute({})
+            task.execute(DummyDag('my_dag', task.task_id).context)
```
Comment (assapin) [inline tests/runners/airflow/tasks/test_python.py:79]: why is this needed?

File: .github/workflows/versioning.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  workflow_run:
+    workflows: ["Running unittest"]
+    branches: [master]
+    types: 
+      - completed
+
+jobs:
+  versioning-auto-commit:
```
Comment (assapin) [inline .github/workflows/versioning.yml:11]: so this will bump the version upon any merge to master.
and it either take a tag or bumps a minor/patch?

**Comment by kaplanlior** [comment]
I wouldn't tag/bump/push on every commit.

Only for those tags with special format (e.g. vX.Y.X)
For CI better to use SHA of the commit.

File: .github/workflows/versioning.yml (hunk 5 above)
Comment (naturalett) [reply .github/workflows/versioning.yml:11]: Yes, any merge to master will produce: **0.0.1rc5-dev1**, **0.0.1rc5-dev2**, **0.0.1rc5-dev3**
It doesn't take a tag. The tag is the release like **0.0.1rc5** and we will add **-dev1**

File: tests/runners/airflow/build/http/python/test_python_server_image_builder.py (hunk 3 above)
Comment (naturalett) [reply tests/runners/airflow/build/http/python/test_python_server_image_builder.py:53]: It doesn't align with what we have in NI.
So either we need it or not. @aviemzur @zionrubin

File: tests/runners/airflow/tasks/test_python.py (hunk 4 above)
Comment (naturalett) [reply tests/runners/airflow/tasks/test_python.py:79]: It was missing. It didn't have a DAG definition to test against it.
@zionrubin

Comment (assapin) [reply tests/runners/airflow/tasks/test_python.py:79]: if the tests were failing, need to take a look. 
if not -> perhaps separate from this commit

**Comment by naturalett** [comment]
> I wouldn't tag/bump/push on every commit.
> 
> Only for those tags with special format (e.g. vX.Y.X)
> For CI better to use SHA of the commit.

Can you elaborate more? Which special tags.
Can you describe a use case?

**Review by assapin — APPROVED** [review]

**Review by assapin — APPROVED** [review]
